tests/app/template/test_rest.py

```diff
@@ -324,6 +325,84 @@ def test_should_get_only_templates_for_that_service(notify_api, sample_user, ser
             assert len(json_resp_4['data']) == 1
 
 
+@pytest.mark.parametrize(
+    "subject, content, path, expected_subject, expected_content, expected_error", [
+        (
+            'about your ((thing))',
+            'hello ((name)) we’ve received your ((thing))',
+            '/service/{}/template/{}',
+            'about your ((thing))',
+            'hello ((name)) we’ve received your ((thing))',
+            None
+        ),
+        (
+            'about your thing',
+            'hello user we’ve received your thing',
+            '/service/{}/template/{}/preview',
+            'about your thing',
+            'hello user we’ve received your thing',
+            None
+        ),
+        (
+            'about your ((thing))',
+            'hello ((name)) we’ve received your ((thing))',
+            '/service/{}/template/{}/preview?name=Amala&thing=document',
+            'about your document',
+            'hello Amala we’ve received your document',
+            None
+        ),
+        (
+            'about your ((thing))',
+            'hello ((name)) we’ve received your ((thing))',
+            '/service/{}/template/{}/preview?eman=Amala&gniht=document',
+            None, None,
+            'Missing personalisation: thing, name'
+        ),
+        (
+            'about your ((thing))',
+            'hello ((name)) we’ve received your ((thing))',
+            '/service/{}/template/{}/preview?name=Amala&thing=document&foo=bar',
+            None, None,
+            'Personalisation not needed for template: foo'
+        )
+    ]
+)
+def test_should_get_a_single_template(
+    notify_api, sample_user, service_factory, subject, content, path, expected_subject, expected_content, expected_error
+):
+    with notify_api.test_request_context(), notify_api.test_client() as client:
+
+        service = service_factory.get('service 1', email_from='service.1')
+
+        template_id = json.loads(client.post(
+            '/service/{}/template'.format(service.id),
+            headers=[('Content-Type', 'application/json'), create_authorization_header()],
+            data=json.dumps({
+                'name': 'my template',
+                'template_type': 'email',
+                'subject': subject,
+                'content': content,
+                'service': str(service.id),
+                'created_by': str(sample_user.id)
+            })
+        ).get_data(as_text=True))['data']['id']
```

Comment:
any reason you're using the endpoint here rather than calling the sample_template fixture as a function? Would rather avoid two fake requests